File: src/main/java/architecture/goldenboughs_lib/mixed/ILivingDamageEvent$Post.java
package architecture.goldenboughs_lib.mixed;

import architecture.goldenboughs_lib.api.NoMixinException;
import net.neoforged.neoforge.common.damagesource.DamageContainer;
import net.neoforged.neoforge.event.entity.living.LivingDamageEvent;

public interface ILivingDamageEvent$Post {
	static ILivingDamageEvent$Post of(LivingDamageEvent.Post obj) {
		return (ILivingDamageEvent$Post) obj;
	}

	default DamageContainer goldenboughs_lib$getDamageContainer() {
		throw new NoMixinException();
	}
}

File: src/main/kotlin/architecture/goldenboughs_lib/mixed/ILivingDamageEvent$Post.kt
package architecture.goldenboughs_lib.mixed

import architecture.goldenboughs_lib.api.NoMixinException
import net.neoforged.neoforge.common.damagesource.DamageContainer
import net.neoforged.neoforge.event.entity.living.LivingDamageEvent

interface `ILivingDamageEvent$Post` {
	fun `goldenboughs_lib$getDamageContainer`(): DamageContainer {
		throw NoMixinException()
	}

	companion object {
		@JvmStatic
		fun of(obj: LivingDamageEvent.Post): `ILivingDamageEvent$Post` {
			return obj as `ILivingDamageEvent$Post`
		}
	}
}
